compiler/rustc_typeck/src/check/coercion.rs

```diff
@@ -216,6 +216,9 @@ impl<'f, 'tcx> Coerce<'f, 'tcx> {
             ty::Ref(r_b, _, mutbl_b) => {
                 return self.coerce_borrowed_pointer(a, b, r_b, mutbl_b);
             }
+            ty::Dynamic(predicates, region, ty::DynStar) if self.tcx.features().dyn_star => {
```

Comment:
One more thing I just thought of, it might be good to have a test case to make sure we don't accidentally coerce to `dyn*` unless the feature is enabled. 

I was chatting with @nikomatsakis a little about this, and he brought up that especially once we have AFIDT it might be possible to have a `dyn*` appear out of thin air due to type inference without anyone explicitly typing it. Although right now I'm not sure we can do that, and I think this check will be enough to keep this from happening on accident.

Reply:
One way to do the test would be to use multiple crates. One would have `pub foo(x: dyn* Sized) {}` , with the feature flag, and another crate would do `base_crate::foo(22)` without the feature flag. That should be a compiler error.

(Thanks to @nikomatsakis for this suggestion)